packages/clerk-js/src/core/clerk.ts

```diff
@@ -831,10 +834,40 @@ export class Clerk implements ClerkInterface {
     if (beforeEmit) {
       beforeUnloadTracker?.startTracking();
       this.#setTransitiveState();
+      deprecated('beforeEmit', 'Use the `redirectUrl` property instead.');
       await beforeEmit(newSession);
       beforeUnloadTracker?.stopTracking();
     }
 
+    if (redirectUrl) {
+      beforeUnloadTracker?.startTracking();
+      this.#setTransitiveState();
+
+      if (
+        this.client.cookieExpiresAt &&
+        this.client.cookieExpiresAt.getTime() - Date.now() <= 8 * 24 * 60 * 60 * 1000 // 8 days
+      ) {
+        const absoluteRedirectUrl = new URL(redirectUrl, window.location.href);
+
+        await this.navigate(
+          this.buildUrlWithAuth(
+            this.#fapiClient
+              .buildUrl({
+                method: 'GET',
+                path: '/client/touch',
+                pathPrefix: 'v1',
+                search: { redirect_url: absoluteRedirectUrl.toString() },
+              })
+              .toString(),
```

Comment:
I agree with the suggestion — my concern here is that it would require introducing a larger refactor to be able to access the methods `buildUrlWithAuth` and the `fapiClient` to be able to construct the URL in the client resource which might grow the surface area of this PR further.

Let me know if you think it makes sense to introduce this change as part of this PR or a later one to keep this one manageable.

Reply:
@issuedat Yeah agreed we can leave `buildUrlWithAuth` as is for now, I _believe_ we can access the fapiClient like this within `Client` so we can construct the FAPI url:

```
BaseResource.getFapiClient()
```

Reply:
Good call! I've used the static method on the `BaseResource` to create a `buildTouchUrl` method on the client which should hopefully further simplify some of this logic in `setActive` as suggested! Thank you 🙇